app/src/main/java/com/google/samples/apps/nowinandroid/navigation/NiaNavHost.kt

```diff
@@ -49,7 +51,13 @@ fun NiaNavHost(
         // TODO: handle topic clicks from each top level destination
         forYouScreen(onTopicClick = {})
         bookmarksScreen(onTopicClick = {})
-        searchScreen(onBackClick = navController::popBackStack)
+        searchScreen(
+            onBackClick = navController::popBackStack,
+            onInterestsClick = { appState.navigateToTopLevelDestination(INTERESTS) },
+            onTopicClick = {
+                navController.navigateToTopic(it)
+            },
```

Comment:
Use method reference to avoid create many lambda (compiler will remember method reference)
```suggestion
            onTopicClick = navController::navigateToTopic,
```

Reply:
Done

Reply:
onTopicClick = {
                navController.navigateToTopic(it)
            },
            onTopicClick = navController::navigateToTopic,